ts/browser-editor/tsconfig.json

```diff
@@ -0,0 +1,11 @@
+{
+    "extends": "../tsconfig.json",
+    "include": ["*"],
+    "references": [
+        { "path": "../lib" },
+        { "path": "../domlib" },
+        { "path": "../sveltelib" },
+        { "path": "../components" },
+        { "path": "../editor" }
+    ]
+}
```

Comment:
Some thoughts:
1. Having more separate modules means we can move out logic from editor.ts. One small example: the logic for `StickyBadge` is only needed for the editor in the AddCards window. However all other editor use cases will also bundle and load it whenever it is invoked. `StickyBadge` is small, so it doesn't matter that much, but they will only get more from here on out: One is the notetype/deck buttons, another is the [card preview](https://github.com/ankitects/anki/pull/969) I've been meaning to implement for a very long time now, which could open in a split pane, but won't be needed in EditCurrent. 
2. I don't intend to increase the amount of modules forever, in fact at some point we could merge modules together: Once I manage to open a previewer in a split pane from the browser and the AddCards editor, we could also open the EditCurrent editor in a split pane from the reviewer, which means that `reviewer-editor` could be merged with the `reviewer` module (hence the name). The same is true for `browser-editor` and a hypothetical `browser` ts module (even though I don't intend to tackle this for a very long time).
3. Is there any way to parallelize module compilation?
4. Would it make sense to try and make `ts/lib` smaller? So that any change in `ts/lib` is guaranteed to actually affect every module. E.g. I'm thinking of `lib/keys` and `lib/shortcuts` here, which are only used in the editor and could be factored out into a `ts/editinglib`.

> do we need esbuild() in ts/editable?

I'm abusing esbuild here to generate the `.css` file, which is included from within editor [here](https://github.com/ankitects/anki/blob/4e4d12a62a261ea90afba88d761a2aad50baf12b/ts/editor/RichTextStyles.svelte#L49). It is kind of a hack, but it's the only way I know to combine the CSS from the Svelte components and the css files together.

Reply:
Will reply below